### ISSUE TITLE
[@visx/shape] A line with gradient stroke doesn't render when it's flat

### ISSUE THREAD

**Issue by Perchyk** [airbnb/visx#1886]
Hello.
We are using LineChart component built with visx and I'm currently enhancing it with tooltip and multiline support.
The design requires to apply a gradient on lines when it's hovered (essentially, it's a fancy version of a dot on a datapoint).
I'm using `<LinePath>` to render the lines.
I have discovered that somehow, if all of the datapoints for a line are equal, it disappears when it had gradient id applied to `stroke` prop. When it uses a color OR one of the datapoints has a different value - even by 0.00001 - it renders as expected.

Please see the sandbox which contains the reproduced issue with a minimal setup:
https://codesandbox.io/p/sandbox/flat-gradiented-line-not-rendering-l8xv3q

I have omitted the tooltip itself because it always renders normally to simplify the example.

There's a fix: as mentioned, increase one of the datapoints values even by 0.0001, and then it renders. (Fun fact: increased by 0.00001, it stops rendering again). You can see that by changing any of the values for `value1` in the `data` array on line 13 in the example.

Library version: latest (though we are on version 2 - so it looks like a long living bug)

**Comment by williaster** [comment]
Hey @Perchyk - this is actually an svg gotcha with gradients. as described in [this stack overflow](https://stackoverflow.com/questions/21638169/svg-line-with-gradient-stroke-wont-display-if-vertical-or-horizontal), 

> From the svg spec
>
> > Keyword objectBoundingBox should not be used when the geometry of the applicable element has no width or no height, such as the case of a horizontal or vertical line, even when the line has actual thickness when viewed due to having a non-zero stroke width since stroke width is ignored for bounding box calculations. When the geometry of the applicable element has no width or height and objectBoundingBox is specified, then the given effect (e.g., a gradient or a filter) will be ignored.
>
> objectBoundingBox is the default for gradientUnits so you need to use `gradientUnits="userSpaceOnUse"` and then adjust the values to be appropriate for the different coordinate system.

**tl;dr** you should set `<LinearGradient {...} gradientUnits="userSpaceOnUse" />`

<img width="300" alt="image" src="https://github.com/user-attachments/assets/79cf4217-afcc-4c28-813a-e841f88944cd">


https://github.com/airbnb/visx/issues/1149 is actually the same. I'll close this to consolidate, but we could consider adding this by default in `@visx/gradient`. if you're interested in that I'm happy to review a PR!
